src/schema/mutation/editApplication.ts

```diff
@@ -73,18 +96,60 @@ export default {
       args.settings && { settings: args.settings },
       args.permissions && { permissions: args.permissions }
     );
+
+    const arrayFilters: any[] = [];
+    if (args.templates) {
+      if (ability.cannot('update', 'Template')) {
+        throw new GraphQLError(errors.permissionNotGranted);
+      }
+      const change: TemplateChange = args.templates;
+
+      // add new template
+      if (change.add) {
+        const pushTemplate = {
+          templates: {
+            name: change.add.name,
+            type: change.add.type,
+            content: change.add.content,
+          },
+        };
+        Object.assign(update, { $addToSet: pushTemplate });
+      }
+
+      // remove template
+      if (change.remove) {
+        Object.assign(update, {
+          $pull: { templates: { _id: change.remove.id } },
+        });
+      }
+
+      // update template
+      if (change.update) {
+        const updateTemplate = {
+          'templates.$[element].name': change.update.name,
+          'templates.$[element].content': change.update.content,
+        };
+
+        Object.assign(update, { $set: updateTemplate });
+        arrayFilters.push({
+          'element._id': new Types.ObjectId(change.update.id),
+        });
+      }
+    }
+
     application = await Application.findOneAndUpdate(filters, update, {
       new: true,
+      arrayFilters,
     });
-    const publisher = await pubsub();
-    publisher.publish('app_edited', {
-      application,
-      user: user.id,
-    });
-    publisher.publish('app_lock', {
-      application,
-      user: user.id,
-    });
+    // const publisher = await pubsub();
+    // publisher.publish('app_edited', {
+    //   application,
+    //   user: user.id,
+    // });
+    // publisher.publish('app_lock', {
+    //   application,
+    //   user: user.id,
+    // });
```

Comment:
@AntoineRelief This bit of code was causing some errors, I'm not entirely sure what is does, so I just commented it out